crates/prices/src/lib.rs

```diff
@@ -0,0 +1,25 @@
+mod chainlink;
+mod error;
+mod feed;
+mod init;
+
+pub use init::init;
+
+use feed::Feed;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+#[derive(Deserialize, Debug)]
+pub struct Feeds {
+    // Structure for feed discovery: network -> currency -> asset
+    feeds: HashMap<String, HashMap<String, HashMap<String, Vec<Feed>>>>,
```

Comment:
the last one should be a `ethers::types::Address`, not a String

Reply:
definitely looks like the best approach, will change it.

Reply:
regarding the `ethers::types::address`, do you mean asset instead? for currency I was considering "USD" and asset I was assuming "ETH", so I guess it's the asset that should be `Address`?

Reply:
I ended up keeping the `String` as it made sense to me using the token symbol instead, as it is in the JSON, to keep the feeds file more readable (token symbol vs address).

Another option would be, while deserializing, to translate the token symbol to its address depending on the network, but I think it's simpler just to access the token metadata and retrieve the symbol before accessing the feed.

Let me know if you disagree!